include/matrix_assembler.h

```diff
@@ -140,6 +92,25 @@ class MatrixAssembler {
   std::shared_ptr<Eigen::SparseMatrix<double>> jac_;
   //! id map that maps node id to variable id
   std::shared_ptr<std::map<Index, Index>> id_map_;
+
+  //! Assemble pressure demand part of the jacobian matrix
+  //! \param[in] n the corresponding node pointer
+  //! \param[in] index the position of the corresponding node in variable vector
+  //! \param[in] update the vector that stores the position of element in
+  //! jacobian matrix
+  void construct_demand_jac(const std::shared_ptr<pipenetwork::Node> & n, Index index,
+                            std::vector<Eigen::Triplet<double>>& update);
```

Comment:
Are we modifying this vector?

Reply:
If you mean update then yes.

Reply:
Can we make it return a vector of update instead?